### PR TITLE
Fix stylistic issues, fix issue #32

### PR DESCRIPTION
**Pull request by vbmade2000**

### CODE REVIEW

**Comment by vbmade2000** [comment]
@jvehent Can we merge this PR and reject other 2 ?

**Comment by jvehent** [comment]
I'd like @mvanotti to have a look before merging, but it looks good to me.

**Comment by vbmade2000** [comment]
Sure

File: common/common_linux.go [hunk 1]
```diff
@@ -7,15 +7,17 @@ import (
 	"strings"
 )
 
+// MapsFilePathFromPid method maps joins "maps" file path with Pid
```
Comment (mvanotti) [inline common/common_linux.go]: I would say that it returns the memory maps file path for a given process id.

File: common/common_linux.go [hunk 2]
```diff
@@ -7,15 +7,17 @@ import (
...
 func MapsFilePathFromPid(pid uint) string {
 	return filepath.Join("/proc", fmt.Sprintf("%d", pid), "maps")
 }
 
+// MemFilePathFromPid method joins "mem" file path with Pid
```
Comment (mvanotti) [inline common/common_linux.go]: Same here. It gives the path of the process' memory file.

File: examples/listlibs.go [hunk 3]
```diff
@@ -54,7 +54,7 @@ func main() {
 
 }
 
-func findProcWithLib(r *regexp.Regexp, ps []process.Process) (matches map[process.Process][]string, harderror error, softerrors []error) {
+func findProcWithLib(r *regexp.Regexp, ps []process.Process) (matches map[process.Process][]string, softerrors []error, harderror error) {
```
Comment (mvanotti) [inline examples/listlibs.go]: Does this even compile? You are changing the order of the parameters, but at the end we are returning the things in the wrong order..

File: listlibs/listlibs.go [hunk 4]
```diff
@@ -7,15 +7,16 @@ import (
...
 	return listLoadedLibraries(p)
 }
 
-// GetMaGetMatchingLoadedLibraries lists the libraries loaded by process p whose path matches r.
-func GetMatchingLoadedLibraries(p process.Process, r *regexp.Regexp) (libraries []string, harderror error,
-	softerrors []error) {
+// GetMatchingLoadedLibraries lists the libraries loaded by process p whose path matches r.
+func GetMatchingLoadedLibraries(p process.Process, r *regexp.Regexp) (libraries []string, softerrors []error,
+	harderror error,
+) {
```
Comment (mvanotti) [inline listlibs/listlibs.go]: Did you run gofmt here? just curious about this formatting..

File: listlibs/listlibs_windows.go [hunk 5]
```diff
@@ -14,11 +14,12 @@ import (
 // #include "listlibs_windows.h"
 import "C"
 
-func listLoadedLibraries(p process.Process) (libraries []string, harderror error, softerrors []error) {
+func listLoadedLibraries(p process.Process) (libraries []string, softerrors []error,
+	harderror error) {
```
Comment (mvanotti) [inline listlibs/listlibs_windows.go]: fomatting problem?

File: memaccess/memaccess.go [hunk 6]
```diff
@@ -19,68 +19,68 @@ func (m MemoryRegion) String() string {
...
+// NoRegionAvailable is a centinel value indicating that there is no more regions available.
 var NoRegionAvailable MemoryRegion
 
 // NextReadableMemoryRegion returns a memory region containing address, or the next readable region after address in
 // case addresss is not in a readable region.
 //
-// If there aren't more regions available the special value NoRegionAvailable is returned.
-func NextReadableMemoryRegion(p process.Process, address uintptr) (region MemoryRegion, harderror error,
-	softerrors []error) {
+// NextReadableMemoryRegion returns next readable memory region. If there aren't more regions available the special
```
Comment (mvanotti) [inline memaccess/memaccess.go]: Isn't this repeating what's on line 25?

File: memaccess/memaccess_c_wrapper.go [hunk 7]
```diff
@@ -13,8 +13,9 @@ import (
 	"unsafe"
 )
 
-func nextReadableMemoryRegion(p process.Process, address uintptr) (region MemoryRegion, harderror error,
-	softerrors []error) {
+func nextReadableMemoryRegion(p process.Process, address uintptr) (region MemoryRegion, softerrors []error,
+	harderror error,
+) {
```
Comment (mvanotti) [inline memaccess/memaccess_c_wrapper.go]: formatting?

File: listlibs/listlibs_windows.go (hunk 5 above)
Comment (vbmade2000) [inline listlibs/listlibs_windows.go]: yes. Solved

File: listlibs/listlibs.go (hunk 4 above)
Comment (vbmade2000) [inline listlibs/listlibs.go]: I did on every file but didn't realize that I should do after modification too. But worked upon it now.

File: examples/listlibs.go (hunk 3 above)
Comment (vbmade2000) [inline examples/listlibs.go]: It was not compiled. Addressed.

**Comment by mvanotti** [comment]
Thanks for the changes :)

**Comment by vbmade2000** [comment]
np. I have learned a lot from this task :)

**Pull request merged**